Система комментариев

licosty opened the pull request:
[Задача JIRA](https://faang-school.atlassian.net/browse/BJS2-83691)

licosty committed:
comment: create, update, delete, findAllByPostId + tests

avzag1 commented on src/main/java/faang/school/postservice/controller/comment/CommentController.java:
```diff
@@ -0,0 +1,44 @@
+    private final CommentService commentService;
+
+    @PostMapping
+    public CommentDto create(@RequestBody @Valid CommentDto commentDto) {
```
Вроде аннотация @Validate, а не @Valid.

licosty replied:
Проверила, @Valid корректно. Но пока проверяла, узнала, что есть @Validated. Она позволяет работать с группами и через эти группы гибко настраивать обязательность валидации полей. Очень познавательно и полезно)

licosty committed:
comment: CommentControllerTest, LikeControllerTest переделаны через @WebMvcTest

zaqqaz226 commented on src/main/java/faang/school/postservice/service/comment/CommentService.java:
```diff
@@ -0,0 +1,97 @@
+                .toList();
+    }
+
+    private void validateAuthor(long currentUserId, long authorId) {
```
Тут либо название надо как то у метода поменять, либо наполнение разделить. А то у тебя написано, что валидация автора, а тут еще и юзера проверяем.

licosty replied:
Выделила проверку наличия юзера в отдельный метод

zaqqaz226 commented on src/main/java/faang/school/postservice/service/comment/CommentService.java:
```diff
@@ -0,0 +1,97 @@
+        }
+    }
+
+    private Post checkPostExists(long id) {
```
Опять же, название говорит, что ты просто хочешь проверить есть он или нет. А по факту ты идешь в базу, забираешь пост и возвращаешь его. Еще и используешь возвращенное ) Что то надо сделать с названием.

licosty replied:
Скорректировала

zaqqaz226 commented on src/main/java/faang/school/postservice/service/comment/CommentService.java:
```diff
@@ -0,0 +1,97 @@
+                () -> new EntityNotFoundException("Post {} not found", id));
+    }
+
+    private Comment checkCommentExists(long id) {
```
тут та же петрушка )

licosty replied:
Скорректировано

zaqqaz226 commented on src/test/java/faang/school/postservice/controller/comment/CommentControllerTest.java:
```diff
@@ -0,0 +1,155 @@
+        return OBJECT_MAPPER.writeValueAsString(obj);
+    }
+
+    private static Stream<Arguments> provideNotValidComment() {
```
Просто придирка, можно не делать. Я предпочитаю, чтобы метод со значениями был прямо над методом теста, откуда будет он дергаться.

zaqqaz226 commented:
Я пытался придираться к структуре, но у меня ничего не получилось)

licosty committed:
comment:
- исправлены имена методов
- логика проверки наличия юзера скорректирована и вынесена из validateAuthor

licosty commented:
Значит будет как похвала)
